### ISSUE TITLE
[FEATURE] Implement AEVA CoreDetection

### ISSUE THREAD

**Issue by pelumitegbe** [oss-slu/mithridatium#61]
## Summary

Implement AEVA-style backdoor detection by running the model on masked versions of clean images and measuring how much the model’s predictions change.
This implementation should:
- use only forward passes
- not inspect weights or gradients
- run on CIFAR-10 using the existing dataloader

## Acceptance Criteria
- [ ] A new file exists: mithridatium/defenses/aeva.py

- [ ] There is a function:
      def run_aeva(model, loader, config) -> dict

- [ ] The function: 
- takes a trained model and CIFAR-10 test loader
- applies random square masks to images
- runs the model on multiple masked versions
- computes a variance-based score
- Returns a dictionary like:
{
  "verdict": "clean" or "suspicious",
  "score": float,
  "num_images": int,
  "num_masks": int
}

## Tasks that need to be completed for this feature

- Create mithridatium/defenses/aeva.py
- Sample a small number of images from loader (e.g. first 100)
-  For each image:
    - create several masked versions (e.g. 10)
    - run model(x) on each version
    - record predicted probabilities
- Compute how much predictions vary across masks
- Average this value into a single AEVA score
- Decide a simple verdict rule (e.g. low variance → suspicious)

## Additional Context

AEVA does not require loading a separate model
Only forward passes are allowed
This is a first-pass implementation, not a perfect reproduction
